File: java/e_58.java
class Solution {
    public int lengthOfLastWord(String s) {
        int end = s.length()-1;

        while (end >= 0 && s.charAt(end) == ' '){
            end--;
        }

        int length = 0;
        while (end >= 0 && s.charAt(end) != ' '){
            length++;
            end--;
        }
        return length;
    }
}

File: rust/e_58.rs
impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let mut s = s.trim();
        let mut length: i32 = 0;
        for c in s.chars().rev() {
            if c != ' ' {
                length += 1;
            } else{
                break;
            }
        }        
        return length;
    }
}